packages/p2p/src/components/my-ads/form-ads.jsx

```diff
@@ -12,22 +12,37 @@ import { textValidator, lengthValidator } from 'Utils/validations';
 import { requestWS } from 'Utils/websocket';
 import AdSummary from './my-ads-summary.jsx';
 
-class FormAds extends React.Component {
-    state = {
-        error_message: '',
-        is_loading: true,
-    };
+const FormAds = ({ handleShowForm }) => {
+    const [advertiser_info, setAdvertiserInfo] = React.useState({});
+    const { currency, local_currency_config } = React.useContext(Dp2pContext);
+    const [error_message, setErrorMessage] = React.useState('');
+    const [is_loading, setIsLoading] = React.useState(true);
+    const { contact_info, default_advert_description, payment_info } = advertiser_info;
+
+    React.useEffect(() => {
+        advertiserInfo();
+    }, []);
 
-    async componentDidMount() {
-        this.setState({
-            is_loading: false,
+    const advertiserInfo = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_info } = response;
+                    setAdvertiserInfo(p2p_advertiser_info);
+                    setIsLoading(false);
```

Comment:
We should also stop loading when there is an error, and that error should be handled.